### ISSUE TITLE
py-udal-interface (from -r requirements.txt (line 8)) does not appear to be a Python project

### ISSUE THREAD

**Issue by ctroupin** [fair-ease/py-udal-fe-impl#7]
Hello,

following the documentation, these instructions

```python
python -m venv .venv
source .venv/bin/activate
cd py-udal-fe-impl
pip install -r requirements.txt
```

lead to the error:

```python
ERROR: file:///home/ctroupin/py-udal-interface (from -r requirements.txt (line 8)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

Is that expected?

**Comment by Marie59** [comment]
Hello, 
+1 on Charles question ! I tried it too and I got the same error...

**Comment by jorgecunhamendes** [comment]
Did you clone also py-udal-interface? The instructions state:

> Clone this repository and [py-udal-interface](https://github.com/fair-ease/py-udal-interface) into the same workspace/directory, [...]

**Comment by ctroupin** [comment]
I don't remember because it was some time ago but I will start from scratch now to see what were the steps.

**Comment by ctroupin** [comment]
Yes I cloned both repos in a common directory, then

```bash
pip install -r requirements.txt
Obtaining file:///home/ctroupin/test2del/py-udal-interface (from -r requirements.txt (line 8))

[notice] A new release of pip is available: 24.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
ERROR: file:///.......py-udal-interface (from -r requirements.txt (line 8)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

**Comment by jorgecunhamendes** [comment]
OK, thank you. I am working on migrating to Poetry for dependency management, which should solve this. I will also try to produce some distributable packages.

**Comment by ctroupin** [comment]
Thanks, don't hesitate to let us know if you need some beta testers, we'll do our best to help

**Comment by jorgecunhamendes** [comment]
I changed the dependency management to use Poetry and it should use py-udal-interface directly from the GitHub repository. I updated the README accordingly. Let me know if that solves it for you.

**Comment by ctroupin** [comment]
OK I've tried what is suggested but it seems I have a problem with poetry:

```bash
poetry install --with notebook

  Stack trace:

  11  /usr/lib/python3/dist-packages/clikit/console_application.py:123 in run
      io = io_factory(

  10  /usr/lib/python3/dist-packages/poetry/console/config/application_config.py:221 in create_io
      resolved_command = application.resolve_command(args)

   9  /usr/lib/python3/dist-packages/clikit/console_application.py:110 in resolve_command
      return self._config.command_resolver.resolve(args, self)

   8  /usr/lib/python3/dist-packages/clikit/resolver/default_resolver.py:34 in resolve
      return self.create_resolved_command(result)

   7  /usr/lib/python3/dist-packages/clikit/resolver/default_resolver.py:166 in create_resolved_command
      if not result.is_parsable():

   6  /usr/lib/python3/dist-packages/clikit/resolver/resolve_result.py:43 in is_parsable
      self._parse()

   5  /usr/lib/python3/dist-packages/clikit/resolver/resolve_result.py:49 in _parse
      self._parsed_args = self._command.parse(self._raw_args)

   4  /usr/lib/python3/dist-packages/clikit/api/command/command.py:113 in parse
      return self._config.args_parser.parse(args, self._args_format, lenient)

   3  /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:53 in parse
      self._parse(args, _fmt, lenient)

   2  /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:101 in _parse
      self._parse_long_option(token, tokens, fmt, lenient)

   1  /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:247 in _parse_long_option
      self._add_long_option(name, None, tokens, fmt, lenient)

  NoSuchOptionException

  The "--with" option does not exist.

  at /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:300 in _add_long_option
      296│     def _add_long_option(
      297│         self, name, value, tokens, fmt, lenient
      298│     ):  # type: (str, Optional[str], List[str], ArgsFormat, bool) -> None
      299│         if not fmt.has_option(name):
    → 300│             raise NoSuchOptionException(name)
      301│ 
      302│         option = fmt.get_option(name)
      303│ 
      304│         if value is False:
```

**Comment by jorgecunhamendes** [comment]
@ctroupin, I am not able to replicate. I have Poetry 1.8.3. I think dependency groups should be available since version 1.2.0. Could you share your Poetry version? Can you try to upgrade if it is older than 1.2.0?

**Comment by ctroupin** [comment]
Thanks!, yes you're right, I had just installed `poetry` and thought I had a recent version but that was not the case.

Now with a recent version (1.8.4) I can now install as expected.
